Kickstarter-iOS/DataSources/CategorySelectionDataSource.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+import KsApi
+import Library
+import UIKit
+
+internal final class CategorySelectionDataSource: ValueCellDataSource {
+  func load(_ categories: [KsApi.Category]) {
+    for (index, parent) in categories.enumerated() {
+      let subcategories = parent.subcategories?.nodes.compactMap { ($0.name, PillCellStyle.grey) } ?? []
+      let allCategories = [("All \(parent.name) Projects", .grey)] + subcategories
+
+      self.set(values: allCategories, cellClass: PillCell.self, inSection: index)
+    }
+  }
+
+  override func configureCell(collectionCell cell: UICollectionViewCell, withValue value: Any) {
+    switch (cell, value) {
+    case let (cell as PillCell, value as (String, PillCellStyle)):
+      cell.configureWith(value: value)
+    default:
+      assertionFailure("Unrecognized (cell, value) combo.")
+    }
+  }
+
+  func collectionView(
+    _ collectionView: UICollectionView,
+    viewForSupplementaryElementOfKind kind: String,
+    at indexPath: IndexPath
+  ) -> UICollectionReusableView {
+    let view = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: "suppView", for: indexPath)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/kickstarter/ios-oss/blob/40776e0be4643b2b756fb688587a0900ce35e49e/Kickstarter-iOS/DataSources/CategorySelectionDataSource.swift#L30"></span>Line should be 110 characters or less: currently 125 characters<br />
<code>line_length</code> <code>CategorySelectionDataSource.swift:30</code></td>
    </tr>
  </tbody>
</table>